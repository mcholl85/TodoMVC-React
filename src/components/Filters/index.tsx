import { NavLink } from 'react-router-dom'

export default function Filters() {
  const activeClassName = 'selected'

  return (
    <ul className='filters'>
      <li>
        <NavLink className={({ isActive }) => (isActive ? activeClassName : undefined)} to='/'>
          All
        </NavLink>
      </li>
      <li>
        <NavLink
          className={({ isActive }) => (isActive ? activeClassName : undefined)}
          to='/active'
        >
          Active
        </NavLink>
      </li>
      <li>
        <NavLink
          className={({ isActive }) => (isActive ? activeClassName : undefined)}
          to='/completed'
        >
          Completed
        </NavLink>
      </li>
    </ul>
  )
}
