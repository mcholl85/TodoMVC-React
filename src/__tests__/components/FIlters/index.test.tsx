import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Filters from '../../../components/Filters'

const ROUTES = [
  { path: '/', name: 'All' },
  { path: '/active', name: 'Active' },
  { path: '/completed', name: 'Completed' },
]

describe('Component Filters', () => {
  test('Matches DOM Snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/']}>
        <Filters />
      </MemoryRouter>,
    )

    expect(asFragment()).toMatchSnapshot()
  })
  test.each`
    path            | name
    ${'/'}          | ${'All'}
    ${'/active'}    | ${'Active'}
    ${'/completed'} | ${'Completed'}
  `('$name link have only active class on $path route', ({ path, name }) => {
    render(
      <MemoryRouter initialEntries={[path]}>
        <Filters />
      </MemoryRouter>,
    )
    const unSelectedRoutes = ROUTES.filter((route) => route.name !== name)
    const navLink = screen.getByRole('link', { name: name })

    unSelectedRoutes.forEach((route) => {
      expect(screen.getByRole('link', { name: route.name })).not.toHaveClass('selected')
    })
    expect(navLink).toHaveClass('selected')
  })
})
