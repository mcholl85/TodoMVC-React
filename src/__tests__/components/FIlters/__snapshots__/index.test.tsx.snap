// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Filters Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <ul
    class="filters"
  >
    <li>
      <a
        aria-current="page"
        class="selected"
        href="/"
      >
        All
      </a>
    </li>
    <li>
      <a
        href="/active"
      >
        Active
      </a>
    </li>
    <li>
      <a
        href="/completed"
      >
        Completed
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
